eng/pipelines/templates/steps/archetype-sdk-docs.yml

```diff
@@ -14,138 +14,19 @@ steps:
-            Add-Content -Path "$(Build.BinariesDirectory)/$($SafeName)/docfx-output/docfx_project/api/index.md" -Value "This Package Contains no Readme."
-            Copy-Item "$(Build.BinariesDirectory)/$($SafeName)/docfx-output/docfx_project/api/index.md" -Destination "$(Build.BinariesDirectory)/$($SafeName)/docfx-output/docfx_project/index.md"-Force
-            Write-Host "Package ReadMe was not found"
-          }
-
-          # Copy over generated yml and other assets
-          Write-Host "Copy over generated yml and other assets"
-          Copy-Item "$($SafeName)/dll-yaml-output/*"-Destination "$($SafeName)/docfx-output/docfx_project/api" -Recurse
-          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/assets/docfx.json" -Destination "$($SafeName)/docfx-output/docfx_project" -Recurse -Force
-          New-Item -Path "$($SafeName)/docfx-output/docfx_project" -Name templates -ItemType directory
-          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/templates/**" -Destination "$(Build.BinariesDirectory)/$($SafeName)/docfx-output/docfx_project/templates" -Recurse -Force
-
-          # Create Toc for Site Navigation
-          Write-Host "Create Toc for Site Navigation"
-          New-Item "$($SafeName)/docfx-output/docfx_project/toc.yml" -Force
-          Add-Content -Path "$($SafeName)/docfx-output/docfx_project/toc.yml" -Value "- name: ${{artifact.name}}`r`n  href: api/`r`n  homepage: api/index.md"
-
-          # Build Doc Content
-          Write-Host "Build Doc Content"
-          ./docfx/docfx.exe build "$($SafeName)/docfx-output/docfx_project/docfx.json"
-
-          # Copy over site Logo
-          Write-Host "Copy over site Logo"
-          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/assets/logo.svg" -Destination "$(Build.BinariesDirectory)/$($SafeName)/docfx-output/docfx_project/_site" -Recurse -Force
-
-        displayName: Build DocFx Documenttion Site for ${{artifact.name}}
-        workingDirectory: $(Build.BinariesDirectory)
-      - task: PublishPipelineArtifact@0
-        condition: succeeded()
-        inputs:
-          artifactName: Docs.${{artifact.name}}
-          targetPath: $(Build.BinariesDirectory)/$(artifactsafename)/docfx-output/docfx_project/_site
\ No newline at end of file
+  - ${{ each artifact in parameters.Artifacts }}:
+    - pwsh: |
+        . $(Build.SourcesDirectory)/doc/ApiDocGeneration/Generate-Api-Docs.ps1
+        Build-Docs -ArtifactName '${{artifact.name}}' -ServiceDirectory '${{parameters.ServiceDirectory}}' `
+        -ArtifactsSafeName '${{artifact.safeName}}' -ArtifactsDirectoryName '${{artifact.directoryName}}' `
+        -LibType '${{parameters.LibType}}' -RepoRoot $(Build.SourcesDirectory) -BinDirectory $(Build.BinariesDirectory)
+      displayName: Build DocFx Documenttion Site for ${{artifact.name}}
+    - task: PublishPipelineArtifact@0
+      condition: succeeded()
+      inputs:
+        artifactName: Docs.${{artifact.name}}
```

Comment:
@scbedd what is the artifact name we use in all the other languages? We should be consistent about them everywhere.

Reply:
```
JS
packages/azure-abort-controller-1.0.1-dev.20191211.1.tgz
documentation/abort-controller.zip

Java
packages/azure-core-1.2.0-beta.1-sources.jar
packages/azure-core-1.2.0-beta.1-javadoc.jar

Python
packages/azure-abort-core-1.1.2.zip
documentation/abort-core.zip

.Net
Docs.Azure.Core
package/Azure.Core.1.1.0.nupkg
```

I think there is significant difference with how the docs are generated and it will be quite a challenge to get it all consistent. For example the doc process up to publish for each package which makes it so I cant keep them all grouped in a `documentation` directory. It can be done but will mean changing a number of things including updates to the doc publishing.

Reply:
Lets file a follow-up issue to figure out the strategy for publishing the docs consistently. It will help actually make the publishing of the docs consistent as well.

Reply:
Issue filled here https://github.com/Azure/azure-sdk/issues/882